src/utils/VatCalculator.js

```diff
@@ -0,0 +1,65 @@
+const TAXBASE_POU = 1 // percentage of unit price
+const TAXBASE_FAPQU = 2 // fixed amount per quantity of unit
+const TAXBASE_FAPIUW = 3 // percentage of base price
+const TAXBASE_PLP = 4 // percentage of labor price
+
+class VatCalcRow {
+  constructor(_basePrice, _qty, _extendedPrice, _baseLaborPrice, _tdPct, _taxDetails) {
+    this.basePrice = _basePrice
+    this.qty = _qty
+    this.extendedPrice = _extendedPrice
+    this.baseLaborPrice = _baseLaborPrice
+    this.vatAmount = 0
+    this.tdPct = _tdPct
+    this.taxDetails = _taxDetails
+  }
+}
+
+const calcVatAmountPerTaxDetail = (vatCalcRow, taxDetail) => {
+  let vatAmount = 0
+
+  switch (taxDetail.taxBase) {
+    case TAXBASE_POU:
+      if (vatCalcRow.tdPct === 0 || vatCalcRow.tdPct === '0') {
```

Comment:
we can use vatCalcRow.tdPct == 0 instead of vatCalcRow.tdPct === 0 || vatCalcRow.tdPct === '0'